From c385b628aa3588aeb6f86f8b98fd3bdb304a296c Mon Sep 17 00:00:00 2001
From: Mohamed Omar Asaker <mohamed.omarasaker@arm.com>
Date: Wed, 11 Jan 2023 12:47:45 +0000
Subject: [PATCH 10/10] Platform: corstone1000:flash_layout simplification

Complete the simplification of the flash layout.
The flash layout contains only the static definitions
that describe the static layout and the boundries of the dynamic
regions.

The dynamic regions addresses are known by the GPT parser.

Signed-off-by: Mohamed Omar Asaker <mohamed.omarasaker@arm.com>
Upstream-Status: Pending [Not submitted to upstream yet]
---
 .../arm/corstone1000/partition/flash_layout.h | 44 -------------------
 1 file changed, 44 deletions(-)

diff --git a/platform/ext/target/arm/corstone1000/partition/flash_layout.h b/platform/ext/target/arm/corstone1000/partition/flash_layout.h
index c5cf94a52c..b01a3621b3 100644
--- a/platform/ext/target/arm/corstone1000/partition/flash_layout.h
+++ b/platform/ext/target/arm/corstone1000/partition/flash_layout.h
@@ -142,28 +142,9 @@
                                              * Host: FIP and Kernel image
                                              */
 
-/****** TODO: START : NEED SIMPLIFICATION BASED ON GPT *******************/
 /* Bank configurations */
 #define BANK_PARTITION_SIZE             (0xFE0000)   /* 15.875 MB */
 #define TFM_PARTITION_SIZE              (0x5E000)    /* 376 KB */
-#define FIP_PARTITION_SIZE              (0x200000)   /* 2 MB */
-#define KERNEL_PARTITION_SIZE           (0xC00000)   /* 12 MB */
-
-/************************************************************/
-/* Bank : Images flash offsets are with respect to the bank */
-/************************************************************/
-
-/* Image 0: BL2 primary and secondary images */
-#define FLASH_AREA_8_ID                 (1)
-#define FLASH_AREA_8_OFFSET             (0) /* starting from 0th offset of the bank */
-#define FLASH_AREA_8_SIZE               (SE_BL2_PARTITION_SIZE)
-
-#define FLASH_AREA_9_ID                 (FLASH_AREA_8_ID + 1)
-#define FLASH_AREA_9_OFFSET             (FLASH_AREA_8_OFFSET + FLASH_AREA_8_SIZE)
-#define FLASH_AREA_9_SIZE               (SE_BL2_PARTITION_SIZE)
-
-#define FWU_METADATA_IMAGE_0_OFFSET     (FLASH_AREA_8_OFFSET)
-#define FWU_METADATA_IMAGE_0_SIZE_LIMIT (FLASH_AREA_8_SIZE + FLASH_AREA_9_SIZE)
 
 /* Macros needed to imgtool.py, used when creating BL2 signed image */
 #define BL2_IMAGE_LOAD_ADDRESS          (SRAM_BASE + TFM_PARTITION_SIZE + BL2_DATA_GAP_SIZE)
@@ -172,33 +153,16 @@
 
 /* Image 1: TF-M primary and secondary images */
 #define FLASH_AREA_0_ID                 (1)
-#define FLASH_AREA_0_OFFSET             (FLASH_AREA_9_OFFSET + \
-                                         FLASH_AREA_9_SIZE)
 #define FLASH_AREA_0_SIZE               (TFM_PARTITION_SIZE)
-
 #define FLASH_AREA_1_ID                 (FLASH_AREA_0_ID + 1)
-#define FLASH_AREA_1_OFFSET             (FLASH_AREA_0_OFFSET + FLASH_AREA_0_SIZE)
 #define FLASH_AREA_1_SIZE               (TFM_PARTITION_SIZE)
 
-#define FWU_METADATA_IMAGE_1_OFFSET     (FLASH_AREA_0_OFFSET)
-#define FWU_METADATA_IMAGE_1_SIZE_LIMIT (FLASH_AREA_0_SIZE + FLASH_AREA_1_SIZE)
-
 /* Image 2: Host FIP */
 #define FIP_SIGNATURE_AREA_SIZE         (0x1000)      /* 4 KB */
 
-#define FLASH_FIP_OFFSET                (FLASH_AREA_1_OFFSET + \
-                                         FLASH_AREA_1_SIZE + FIP_SIGNATURE_AREA_SIZE)
-#define FLASH_FIP_ADDRESS               (FLASH_BASE_ADDRESS + FLASH_FIP_OFFSET)
-#define FLASH_FIP_SIZE                  (FIP_PARTITION_SIZE)
-
 /* Host BL2 (TF-A) primary and secondary image. */
 #define FLASH_AREA_2_ID                 (FLASH_AREA_1_ID + 1)
 #define FLASH_AREA_3_ID                 (FLASH_AREA_2_ID + 1)
-#define FLASH_INVALID_OFFSET            (0xFFFFFFFF)
-#define FLASH_INVALID_SIZE              (0xFFFFFFFF)
-
-#define FWU_METADATA_IMAGE_2_OFFSET     (FLASH_FIP_OFFSET)
-#define FWU_METADATA_IMAGE_2_SIZE_LIMIT (FLASH_FIP_SIZE)
 
 /* Macros needed to imgtool.py, used when creating TF-M signed image */
 #define S_IMAGE_LOAD_ADDRESS            (SRAM_BASE)
@@ -216,14 +180,6 @@
 
 #define FLASH_AREA_IMAGE_SCRATCH        255
 
-/* Image 3: Kernel image */
-#define KERNEL_PARTITION_OFFSET         (FLASH_FIP_OFFSET + FLASH_FIP_SIZE)
-
-#define FWU_METADATA_IMAGE_3_OFFSET     (KERNEL_PARTITION_OFFSET)
-#define FWU_METADATA_IMAGE_3_SIZE_LIMIT (KERNEL_PARTITION_SIZE)
-
-/****** TODO: END : NEED SIMPLIFICATION BASED ON GPT *******************/
-
 /*******************************/
 /*** ITS, PS and NV Counters ***/
 /*******************************/
-- 
2.25.1

